web/src/js/ducks/modes/regular.ts

```diff
@@ -0,0 +1,64 @@
+import {
+    RECEIVE as RECEIVE_OPTIONS,
+    UPDATE as UPDATE_OPTIONS,
+} from "../options";
+import { ModeState, updateMode } from "./utils";
+import { addListenAddr, getModesOfType } from "./utils";
+
+export const MODE_REGULAR_TOGGLE = "MODE_REGULAR_TOGGLE";
+
+interface RegularState extends ModeState {}
+
+export const initialState: RegularState = {
+    active: true,
+    name: "regular",
+};
+
+export const getMode = (modes) => {
+    const regularMode = modes.regular;
+    return addListenAddr(regularMode);
+};
+
+export const toggleRegular =
+    (updateModeFunc = updateMode) =>
+    async (dispatch) => {
+        dispatch({ type: MODE_REGULAR_TOGGLE });
+
+        const result = await dispatch(updateModeFunc());
+
+        if (!result.success) {
+            // TODO: handle error
+        }
+    };
+
+const regularReducer = (state = initialState, action): RegularState => {
+    switch (action.type) {
+        case MODE_REGULAR_TOGGLE:
+            return {
+                ...state,
+                active: !state.active,
+            };
+        case UPDATE_OPTIONS:
+        case RECEIVE_OPTIONS:
+            if (action.data && action.data.mode) {
+                const currentModeConfig = getModesOfType(
+                    "regular",
+                    action.data.mode.value,
+                )[0];
+                const isActive = currentModeConfig !== undefined;
+                return {
+                    ...state,
+                    active: isActive,
+                    listen_host:
+                        currentModeConfig?.listen_host || state.listen_host,
```

Comment:
This here is still bugged (and uncaught by tests): If we configure an alternative listen port and then receive an update with just `regular`, that change will not be reflected in the UI. (Same type of issue as we had with local mode).

Reply:
Could you please explain it better please?

Reply:
Could you please explain it better please? (btw the logic to change port and host hasn't been implemented yet since we said to start with a single toggle for the regular)

Reply:
Also related to this and to the local mode:
Let's say we write `curl` in the applications field in the **Capture tab**, without activating the local mode. Then we go in the **options modal** and we write `local`. Should we activate `local:curl` or just `local`? At the moment it's the second case. Same thing for the regular mode with the `listen_host` and the `listen_port`. (For the future, does having a mode field in the option modal still make sense once we'll have a dedicated page? 🤔 )

Reply:
> Then we go in the **options modal** and we write `local`. Should we activate `local:curl` or just `local`?

Just local. If we get an update from the server with mode X activated, we overwrite our UI state for mode X. If we get a mode update from the server where mode X is not active, we can keep our UI state as it's deactivated anyhow. Does that make sense?

Reply:
okay makes sense :)